### PR TITLE
Use obsolete jsoncpp deb package instead of script where possible

### PR DESCRIPTION
**Pull request by Marenz**

### CODE REVIEW

**Comment by Marenz** [comment]
This seems to decrease the build dependency install step from 52 seconds to 9 seconds, at least in one sample I checked. Might be a fluke of course.

**Comment by chriseth** [comment]
You were not able to replace it in one of the tests, is that right?

**Comment by Marenz** [comment]
Yeah, the macos one. Maybe there is a release somewhere for that too, but I don't know mac enough to tell.

File: .circleci/config.yml [hunk 1]
```diff
@@ -134,8 +134,7 @@ jobs:
           name: Install build dependencies
           command: |
             apt-get -qq update
-            apt-get -qy install cmake libboost-regex-dev libboost-filesystem-dev libboost-test-dev libboost-system-dev libboost-program-options-dev libcvc4-dev
-            ./scripts/install_obsolete_jsoncpp_1_7_4.sh
+            apt-get -qy install cmake libboost-regex-dev libboost-filesystem-dev libboost-test-dev libboost-system-dev libboost-program-options-dev libcvc4-dev libjsoncpp-dev=1.7.4-3
```
Comment (ekpyron) [inline .circleci/config.yml]: Is there a way to specify the package version ``1.7.4`` without explicitly specifying the build version ``-3`` as well? That'd be nicer - but probably fine either way - I don't expect it to change a lot.

**Review by ekpyron — DISMISSED** [review]
Would be nicer to get rid of the script altogether, but I'd suggest to keep it for the macos build for now and remove it, once we got rid of the whole build-json-cpp-on-our-own-mess in general (then we can just use 1.8.4 from homebrew instead - they in fact have it: https://formulae.brew.sh/formula/jsoncpp).

File: .circleci/config.yml (hunk 1 above)
Comment (Marenz) [reply .circleci/config.yml]: I pushed a wildcard version, let's see if that works

Comment (Marenz) [reply .circleci/config.yml]: it worked :)

**Review by ekpyron — APPROVED** [review]

**Pull request merged**